mongodb/activity/insertDocument/activity.go

```diff
@@ -0,0 +1,184 @@
+
+	dataJSONByte, err := json.Marshal(inputVal.Data)
+	if err != nil {
+		return false, fmt.Errorf("Error reading Data json %s", err.Error())
+	}
+	inputJSON = string(dataJSONByte)
+	logInsert.Debugf("Insert Data JSON from input: ", inputJSON)
+	if inputJSON == "" || inputJSON == "null" {
+		return false, fmt.Errorf("Input Data cannot be null ")
+	}
+
+	db := a.client.Database(a.database)
+	coll := db.Collection(a.collectionName)
+	timeout := a.timeout
+	if timeout == 0 {
+		timeout = 60 //set a default timeout of 60 seconds if no timeout is specified
+	}
+	cntx, cancel := ctx.WithTimeout(ctx.Background(), time.Duration(timeout)*time.Second)
+	defer cancel()
+	failedNum := 0
+	succeedNum := 0
+	totalCount := 0
+	logInsert.Debugf("Insert Operation to perform is [%s] ", a.operation)
+	if a.operation == "Insert One Document" {
+		m := make(map[string]interface{})
+		err = json.Unmarshal([]byte(inputJSON), &m)
+		if err != nil {
+			return false, err
+		}
+		result, err := coll.InsertOne(cntx, m)
+		if err != nil {
+			return false, err
+		}
+		context.SetOutput("insertedId", result.InsertedID)
+	} else {
+		insertOptions := *options.InsertMany()
+		// SetOrdered configures the ordered option. If true, when a write fails, the function will return without attempting
+		// remaining writes. Defaults to true.
+		if a.contErr {
+			logInsert.Debug("Continue on Error is set to true")
+			insertOptions.SetOrdered(false)
+		} else {
+			logInsert.Debug("Continue on Error is set to false")
+		}
+
+		var m []interface{}
+		err = json.Unmarshal([]byte(inputJSON), &m)
+		if err != nil {
+			return false, err
+		}
+
+		result, err := coll.InsertMany(cntx, m, &insertOptions)
+		totalCount = len(m)
+		if err != nil {
+			msg := err.Error()
+			if a.contErr {
+				if strings.Contains(msg, "E1100") {
+					logInsert.Warnf(
+						"Duplicate Key Error occurred while performing Inserting Many Documents."+
+							" Since Continue on Error is set to True, error is being handled gracefully."+
+							" Original Error for reference : [%s] ", msg)
+					for strings.Contains(msg, "E1100") {
+						failedNum = failedNum + 1
+						index := strings.Index(msg, "E11000") + 6
+						msg = msg[index:len(msg)]
+					}
+
+					succeedNum = totalCount - failedNum
+				} else {
+					logInsert.Errorf("Unexpected Error while performing Insert Many Documents Operation : [%s] ", msg)
+					return false, err
+				}
+			} else {
+				/* TODO: Handle this scenario better
+				Since API does not properly return list of InsertedIDs in case of continue on Error = false
+				and there was an error in one of the inserts, there is no good way to handle this case.
+				Users will see incorrect data in the output.
+				*/
+				logInsert.Warn(
+					"Error encountered during Insert Many Documents with Continue on Error set to false."+
+						"Users should check the mongoDB backend for the result of this particular execution for clarity."+
+						"Original error for reference : [%s] ", msg)
+			}
```

Comment:
You can create a function to handle this.

Reply:
@skothari-tibco I would like to have this piece of code within the eval itself. The number of lines of code is not much and it's a bit exaggerated by the lengthy log messages I am printing out for each case. I feel it's better to keep it within the eval